### DISCUSSION TITLE
Add Selector Style Dynamically

### DISCUSSION THREAD

**Discussion by trinhhoanglong2000** [GrapesJS/grapesjs#4235] [Q&A] [answered]
Hi, is it possible to add Selector dynamically?
For example, i have this line of code: component.setStyle({ "text-align": "center" });
which will generate a css 
#iy1ak{
  text-align:center;
 }
 where #iy1ak is the id of that component
 now i want to add selector like : #iy1ak:after or maybe #iy1ak:hover dynamically
 Is there a way to do so ?
 Preciate the helps

**Comment by artf** [comment, marked as answer]
https://grapesjs.com/docs/api/css_composer.html#setrule
